File: app/src/main/java/com/martinrevert/latorrentola/model/YTS/Torrent.java
package com.martinrevert.latorrentola.model.YTS;

/**
 * Created by martin on 22/11/17.
 */

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Torrent {

    @SerializedName("url")
    @Expose
    private String url;
    @SerializedName("hash")
    @Expose
    private String hash;
    @SerializedName("quality")
    @Expose
    private String quality;
    @SerializedName("seeds")
    @Expose
    private Integer seeds;
    @SerializedName("peers")
    @Expose
    private Integer peers;
    @SerializedName("size")
    @Expose
    private String size;
    @SerializedName("size_bytes")
    @Expose
    private String sizeBytes;
    @SerializedName("date_uploaded")
    @Expose
    private String dateUploaded;
    @SerializedName("date_uploaded_unix")
    @Expose
    private Integer dateUploadedUnix;

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getHash() {
        return hash;
    }

    public void setHash(String hash) {
        this.hash = hash;
    }

    public String getQuality() {
        return quality;
    }

    public void setQuality(String quality) {
        this.quality = quality;
    }

    public Integer getSeeds() {
        return seeds;
    }

    public void setSeeds(Integer seeds) {
        this.seeds = seeds;
    }

    public Integer getPeers() {
        return peers;
    }

    public void setPeers(Integer peers) {
        this.peers = peers;
    }

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getSizeBytes() {
        return sizeBytes;
    }

    public void setSizeBytes(String sizeBytes) {
        this.sizeBytes = sizeBytes;
    }

    public String getDateUploaded() {
        return dateUploaded;
    }

    public void setDateUploaded(String dateUploaded) {
        this.dateUploaded = dateUploaded;
    }

    public Integer getDateUploadedUnix() {
        return dateUploadedUnix;
    }

    public void setDateUploadedUnix(Integer dateUploadedUnix) {
        this.dateUploadedUnix = dateUploadedUnix;
    }

}

File: app/src/main/java/com/martinrevert/latorrentola/model/YTS/Torrent.kt
package com.martinrevert.latorrentola.model.YTS

import com.google.gson.annotations.SerializedName
import kotlinx.serialization.Serializable

@Serializable
data class Torrent(
    @SerializedName("url")
    val url: String? = null,
    @SerializedName("hash")
    val hash: String? = null,
    @SerializedName("quality")
    val quality: String? = null,
    @SerializedName("seeds")
    val seeds: Int? = null,
    @SerializedName("peers")
    val peers: Int? = null,
    @SerializedName("size")
    val size: String? = null,
    @SerializedName("size_bytes")
    val sizeBytes: String? = null,
    @SerializedName("date_uploaded")
    val dateUploaded: String? = null,
    @SerializedName("date_uploaded_unix")
    val dateUploadedUnix: Int? = null,
    @SerializedName("type")
    val type: String? = null
)
